src/Datadog.Trace.ClrProfiler.Native/integration_loader.cpp

```diff
@@ -110,55 +103,90 @@ std::vector<Integration> LoadIntegrationsFromStream(std::istream& stream)
             Logger::Warn("Failed to load integrations: ", ex.what());
         }
     }
-
-    return integrations;
 }
 
 namespace
 {
 
-    std::pair<Integration, bool> IntegrationFromJson(const json::value_type& src)
+    void IntegrationFromJson(const json::value_type& src, std::vector<IntegrationMethod>& integrationMethods,
+                             const bool isCallTargetEnabled, const bool isNetstandardEnabled,
+                             const std::vector<WSTRING> disabledIntegrationNames)
     {
         if (!src.is_object())
         {
-            return std::make_pair<Integration, bool>({}, false);
+            return;
         }
 
         // first get the name, which is required
-        const auto name = ToWSTRING(src.value("name", ""));
+        const WSTRING name = ToWSTRING(src.value("name", ""));
         if (name.empty())
         {
             Logger::Warn("Integration name is missing for integration: ", src.dump());
-            return std::make_pair<Integration, bool>({}, false);
+            return;
+        }
+
+        // check if the integration is disabled
+        for (const WSTRING disabledName : disabledIntegrationNames)
+        {
+            if (name == disabledName)
+            {
+                return;
+            }
         }
 
-        std::vector<MethodReplacement> replacements;
         auto arr = src.value("method_replacements", json::array());
         if (arr.is_array())
         {
             for (auto& el : arr)
             {
-                auto mr = MethodReplacementFromJson(el);
-                if (std::get<1>(mr))
-                {
-                    replacements.push_back(std::get<0>(mr));
-                }
+                MethodReplacementFromJson(el, name, integrationMethods, isCallTargetEnabled, isNetstandardEnabled);
             }
         }
-        return std::make_pair<Integration, bool>({name, replacements}, true);
     }
 
-    std::pair<MethodReplacement, bool> MethodReplacementFromJson(const json::value_type& src)
+    void MethodReplacementFromJson(const json::value_type& src, const WSTRING integrationName,
```

Comment:
could you pass `integrationName` by `const &` intead?

Reply:
done in 1aa6b9d